### PR TITLE
visual studio 2022/2019 build instructions

### PR DESCRIPTION
**Pull request by ab9rf**
update instructions to recommend using VS 2022 or VS 2019 with downlevel tool support

### CODE REVIEW

**Comment by myk002** [comment]
Should we mark this as fixing #1743 (and #1480)?

File: docs/Compile.rst [hunk 1]
```diff
@@ -469,7 +469,8 @@ Dependencies
 ------------
 You will need the following:
 
-* Microsoft Visual C++ 2015 or 2017
+* Microsoft Visual C++ 2022, 2019, 2017, or 2015 (optional)
```
Comment (myk002) [inline docs/Compile.rst:472]: Instead of "optional", how about "if editing code"?

**Review by myk002 — APPROVED** [review]

**Comment by myk002** [comment]
Needs changelog entry

File: docs/Compile.rst (hunk 1 above)
Comment (ab9rf) [reply docs/Compile.rst:472]: it's a perfectly reasonable option to use VS Code to edit code and let VS Code call the build tools. the integration isn't even all that bad.

Comment (myk002) [reply docs/Compile.rst:472]: Ah, ok. I wasn't aware that VS Code and Visual C++ were different products. The last time I did Windows programming, there was just Visual Studio, and Visual C++ just plugged into that.

**Comment by myk002** [comment]
Does this PR fully cover the "our build scripts work with VS2015, but VS2017-19 require different generator names and architecture flags passed to CMake" concern from #1743?

**Pull request merged**